### PR TITLE
Added different types of fixed interventions to optimize

### PR DESCRIPTION
**Pull request by anirban-chaudhuri**
`optimize` can now handle fixed interventions on:
- static parameter interventions (already existed)
- static state interventions
- dynamic parameter interventions
- dynamic state interventions
Note that the fixed dynamic interventions might compromise the optimizer's performance so extra care should be taken when using them.

### CODE REVIEW

File: docs/source/interfaces.ipynb
Comment (SamWitty) [inline docs/source/interfaces.ipynb:1]: Can you remove this file from this PR? The only change in this file is metadata from running it with a different virtual environment.

File: pyciemss/interfaces.py [hunk 1]
```diff
@@ -627,13 +627,13 @@ def calibrate(
...
             - By default we set the `start_time` to be 0.
-        static_state_interventions: Dict[float, Dict[str, Intervention]]
+        static_state_interventions: Dict[torch.Tensor, Dict[str, Intervention]]
             - A dictionary of static interventions to apply to the model.
             - Each key is the time at which the intervention is applied.
             - Each value is a dictionary of the form {state_variable_name: intervention_assignment}.
             - Note that the `intervention_assignment` can be any type supported by
               :func:`~chirho.interventional.ops.intervene`, including functions.
-        static_parameter_interventions: Dict[float, Dict[str, Intervention]]
+        static_parameter_interventions: Dict[torch.Tensor, Dict[str, Intervention]]
```
Comment (SamWitty) [inline pyciemss/interfaces.py:636]: Is this change necessary for this PR or is it separate?

**Review by SamWitty — APPROVED** [review]
Approving despite my minor housekeeping comments.

File: pyciemss/interfaces.py (hunk 1 above)
Comment (anirban-chaudhuri) [reply pyciemss/interfaces.py:657]: It is a correction. There is a mismatch between what is in the function definition and the documentation. For this small of a change, I think it is better to fix it now.

File: docs/source/interfaces.ipynb
Comment (anirban-chaudhuri) [reply docs/source/interfaces.ipynb:1]: I merged main into this branch but the metadata changes remained. Not exactly sure how to undo the metadata changes. I am going to go ahead and merge this PR since it will not make any difference to the file currently on main.
